tests/serde.rs

```diff
@@ -0,0 +1,352 @@
+            hash256_preimages: vec![(sha256d::Hash::hash(&[]), vec![1, 2])].into_iter().collect(),
+            proprietary: proprietary.clone(),
+            unknown: unknown.clone(),
+            ..Default::default()
+        }],
+        outputs: vec![Output {
+            bip32_derivation: keypaths.into_iter().collect(),
+            proprietary: proprietary.clone(),
+            unknown: unknown.clone(),
+            ..Default::default()
+        }],
+    };
+
+    let got = serialize(&psbt).unwrap();
+    let want = include_bytes!("data/serde/partially_signed_transaction_bincode") as &[_];
+    assert_eq!(got, want)
+}
+
+#[test]
+fn serde_regression_raw_pair() {
+    let pair = Pair {
+        key: Key {
+            type_value: 1u8,
+            key: vec![0u8, 1u8, 2u8, 3u8],
+        },
+        value: vec![0u8, 1u8, 2u8, 3u8],
+    };
+    let got = serialize(&pair).unwrap();
+    let want = include_bytes!("data/serde/raw_pair_bincode");
+    assert_eq!(got, want)
+}
+
+#[test]
+fn serde_regression_proprietary_key() {
+    let key = ProprietaryKey {
+        prefix: vec![0u8, 1u8, 2u8, 3u8],
+        subtype: 1u8,
+        key: vec![0u8, 1u8, 2u8, 3u8],
+    };
+    let got = serialize(&key).unwrap();
+    let want = include_bytes!("data/serde/proprietary_key_bincode");
+    assert_eq!(got, want)
+}
+
+#[test]
+fn serde_regression_schnorr_sig() {
+    let s = include_str!("data/serde/schnorr_sig_hex");
+    let sig = SchnorrSig {
+        sig: secp256k1::schnorr::Signature::from_str(s.trim()).unwrap(),
+        hash_ty: SchnorrSigHashType::All,
+    };
+
+    let got = serialize(&sig).unwrap();
+    let want = include_bytes!("data/serde/schnorr_sig_bincode") as &[_];
+    assert_eq!(got, want)
+}
+
+#[test]
+fn serde_regression_taproot_builder() {
+    let ver = LeafVersion::from_consensus(0).unwrap();
+    let script = Script::from(vec![0u8, 1u8, 2u8]);
+    let builder = TaprootBuilder::new().add_leaf_with_ver(1, script, ver).unwrap();
+
+    let got = serialize(&builder).unwrap();
+    let want = include_bytes!("data/serde/taproot_builder_bincode") as &[_];
+    assert_eq!(got, want)
+}
+
+#[test]
+fn serde_regression_uint256() {
+    let u = Uint256::from_u64(0xDEADBEEFDEADBEEF).unwrap();
+    let got = serialize(&u).unwrap();
+    let want = include_bytes!("data/serde/uint256_bincode") as &[_];
+    assert_eq!(got, want)
+}
+
+#[test]
+fn serde_regression_uint128() {
+    let u = Uint128::from_u64(0xDEADBEEFDEADBEEF).unwrap();
+    let got = serialize(&u).unwrap();
+    let want = include_bytes!("data/serde/uint128_bincode");
+    assert_eq!(got, want)
+}
+
+/// Helper function to create binary data from debug output.
+///
+/// Reads a comma separated list of integers from file `/tmp/<name>` into a vector and writes it as
+/// binary data to `/tmp/<name>_bincode`. The integers represent the expected bincode for a
+/// serialized data type.
+#[allow(dead_code)]
```

Comment:
I really don't like having this in tests, could you move it to a separate tool?

Reply:
Yeah sure thing, I was kind of feeling the same when I committed it but couldn't think of a better place. Will think a bit harder. Thanks

Reply:
I added a file `util.sh`, put the function in there, improved it a bit, and added documentation on its use to the top of `serde.rs`.